src/sentry/static/sentry/app/components/idBadge/badgeDisplayName.jsx

```diff
@@ -0,0 +1,12 @@
+import styled from 'react-emotion';
+
+import overflowEllipsis from 'app/styles/overflowEllipsis';
+import space from 'app/styles/space';
+
+const BadgeDisplayName = styled('span')`
+  ${overflowEllipsis};
+  max-width: ${p => p.theme.settings.maxCrumbWidth};
```

Comment:
This seems like a concern we don't want to worry about in every single badge.

Can you explain the problem here a little more? Maybe I can help suggest some flex-fu to move this up to `breadcrumb` somehow.

Reply:
So here are some examples of where this is used:
![image](https://user-images.githubusercontent.com/79684/39599856-dc3126b2-4ed1-11e8-8976-07cf9586c9f5.png)

I was debating where this should be handled, and I think the badge makes sense (especially if we start using it everywhere we want to display a name/badge). User+other data slugs are generally the only user input that we echo back to the user, so it has the potential to disrupt layouts when not accounted for.

It'd be preferable if this wasn't a theme property and was handled by a prop though.